components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/ConsentMgtPostAuthnHandler.java

```diff
@@ -185,14 +187,28 @@ protected PostAuthnHandlerFlowStatus handlePreConsent(HttpServletRequest request
 
                 if (isDebugEnabled()) {
                     String message = "Require consent for mandatory claims: %s, requested claims: %s, from user: %s " +
-                                     "for service provider: %s in tenant domain: %s.";
+                            "for service provider: %s in tenant domain: %s.";
 
                     message = String.format(message, consentClaimsData.getMandatoryClaims(),
-                                            consentClaimsData.getRequestedClaims(),
-                                            authenticatedUser.getAuthenticatedSubjectIdentifier(),
-                                            serviceProvider.getApplicationName(), getSPTenantDomain(serviceProvider));
+                            consentClaimsData.getRequestedClaims(),
+                            authenticatedUser.getAuthenticatedSubjectIdentifier(),
+                            serviceProvider.getApplicationName(), getSPTenantDomain(serviceProvider));
                     logDebug(message);
                 }
+                if (LoggerUtils.isDiagnosticLogsEnabled()) {
+                    Map<String, Object> params = new HashMap<>();
+                    params.put(FrameworkConstants.LogConstants.MANDATORY_CLAIMS,
+                            consentClaimsData.getMandatoryClaims());
+                    params.put(FrameworkConstants.LogConstants.REQUESTED_CLAIMS,
+                            consentClaimsData.getRequestedClaims());
+                    params.put(FrameworkConstants.LogConstants.USER,
+                            authenticatedUser.getAuthenticatedSubjectIdentifier());
+                    params.put(FrameworkConstants.LogConstants.SERVICE_PROVIDER, serviceProvider.getApplicationName());
+                    params.put(FrameworkConstants.LogConstants.TENANT_DOMAIN, getSPTenantDomain(serviceProvider));
+                    LoggerUtils.triggerDiagnosticLogEvent(FrameworkConstants.LogConstants.AUTHENTICATION_FRAMEWORK,
+                            params, LogConstants.SUCCESS, "Require consent for mandatory claims ",
```

Comment:
Is this only for mandatory claims? or all claims

Reply:
```
{"logId":"8b262f48-f298-4192-8d73-4605bb6a4c5e","recordedAt":{"seconds":1668778734,"nanos":807590000},"requestId":"7248fd35-072f-49c0-b64c-cf2ce1e52776","resultStatus":"SUCCESS","resultMessage":"Require consent for mandatory claims ","actionId":"process-claim-consent","componentId":"authentication-framework","input":{"tenant domain":"risekite.com","service provider":"sample-app","requested claims":[{"id":0,"claimUri":"http://wso2.org/claim/country","displayName":"Country","description":"Country"}],"user":"aca7acd9-5b7d-49d1-9757-308352aa2072","mandatory claims":[]}}
```

Reply:
Then can we remove the mandatory part in the log message?

Reply:
+1. removed the "mandatory" part with https://github.com/wso2/carbon-identity-framework/pull/4326/commits/9a1423aa5dca4a529cd5a0abe0c755ac81eee53c